File: Php/app/Urls/CashgameFactsUrlModel.php
namespace app\Urls{

	use app\RouteFormats;
	use app\Urls\BaseClasses\HomegameYearUrlModel;
	use entities\Homegame;

	class CashgameFactsUrlModel extends HomegameYearUrlModel{

		public function __construct(Homegame $homegame, $year){
			parent::__construct(RouteFormats::cashgameFacts, RouteFormats::cashgameFactsWithYear, $homegame, $year);
		}

	}

}

File: Web/Models/UrlModels/CashgameFactsUrlModel.cs
using Web.Routing;

namespace Web.Models.UrlModels
{
    public class CashgameFactsUrlModel : HomegameWithOptionalYearUrlModel
    {
        public CashgameFactsUrlModel(string slug, int? year)
            : base(RouteFormats.CashgameFacts, RouteFormats.CashgameFactsWithYear, slug, year)
        {
        }
    }
}